src/diffusers/loaders/unet.py

```diff
@@ -834,6 +845,67 @@ def _convert_ip_adapter_attn_to_diffusers(self, state_dicts, low_cpu_mem_usage=F
                 )
                 attn_procs[name] = attn_processor_class()
+
+                for state_dict in state_dicts:
+                    if f"{key_id}.to_k_lora.down.weight" in state_dict["ip_adapter"]:
+                        lora_dict.update(
+                            {
+                                f"unet.{name}.to_k_lora.down.weight": state_dict["ip_adapter"][
+                                    f"{key_id}.to_k_lora.down.weight"
+                                ]
+                            }
+                        )
+                        lora_dict.update(
+                            {
+                                f"unet.{name}.to_q_lora.down.weight": state_dict["ip_adapter"][
+                                    f"{key_id}.to_q_lora.down.weight"
+                                ]
+                            }
+                        )
+                        lora_dict.update(
+                            {
+                                f"unet.{name}.to_v_lora.down.weight": state_dict["ip_adapter"][
+                                    f"{key_id}.to_v_lora.down.weight"
+                                ]
+                            }
+                        )
+                        lora_dict.update(
+                            {
+                                f"unet.{name}.to_out_lora.down.weight": state_dict["ip_adapter"][
+                                    f"{key_id}.to_out_lora.down.weight"
+                                ]
+                            }
+                        )
+                        lora_dict.update(
+                            {
+                                f"unet.{name}.to_k_lora.up.weight": state_dict["ip_adapter"][
+                                    f"{key_id}.to_k_lora.up.weight"
+                                ]
+                            }
+                        )
+                        lora_dict.update(
+                            {
+                                f"unet.{name}.to_q_lora.up.weight": state_dict["ip_adapter"][
+                                    f"{key_id}.to_q_lora.up.weight"
+                                ]
+                            }
+                        )
+                        lora_dict.update(
+                            {
+                                f"unet.{name}.to_v_lora.up.weight": state_dict["ip_adapter"][
+                                    f"{key_id}.to_v_lora.up.weight"
+                                ]
+                            }
+                        )
+                        lora_dict.update(
+                            {
+                                f"unet.{name}.to_out_lora.up.weight": state_dict["ip_adapter"][
+                                    f"{key_id}.to_out_lora.up.weight"
+                                ]
+                            }
+                        )
+                        key_id += 1
```

Comment:
are there more than one face-id checkpoints right now? does it make sense for us to support more than one?

Reply:
Face ID and Face ID XL are both supported by this PR
Face ID Plus models have different image projection layers